### ISSUE TITLE
The progress bar is animated when 0 rules are selected

### ISSUE THREAD

**Issue by mpreisler** [OpenSCAP/scap-workbench#2]
It goes back and forth. Ideally it would still display 0% before Scan is pressed and 100% after scan of 0 rules has completed.

This issue is specific to the upcoming 1.1.0 release.

Original trac ticket: https://fedorahosted.org/scap-workbench/ticket/283

**Comment by mpreisler** [comment]
This is quite easy to achieve but there is a significant UX problem: The progress bar label says how many rules are selected and it will say 1 rule is selected even though 0 rules are selected. I think this is an issue that is more severe than the animated progress bar.

The patch:
```
diff --git a/src/MainWindow.cpp b/src/MainWindow.cpp
index df96721..7cd8a5c 100644
--- a/src/MainWindow.cpp
+++ b/src/MainWindow.cpp
@@ -236,7 +236,7 @@ void MainWindow::clearResults()
     if (mScanningSession && mScanningSession->fileOpened())
     {                                                                                                                                                                                         
         struct xccdf_policy* policy = mScanningSession != 0 ? xccdf_session_get_xccdf_policy(mScanningSession->getXCCDFSession()) : 0;                                                        
-        mUI.progressBar->setRange(0, policy ? xccdf_policy_get_selected_rules_count(policy) : 1);                                                                                             
+        mUI.progressBar->setRange(0, policy ? std::max(xccdf_policy_get_selected_rules_count(policy), 1) : 1);                                                                                
     }                                                                                                                                                                                         
     else                                                                                                                                                                                      
     {                                                                                                                                                                                         
@@ -464,7 +464,7 @@ void MainWindow::scanAsync(ScannerMode scannerMode)                                                                                                                        
         return;                                                                                                                                                                               
     }                                                                                                                                                                                         
                                                                                                                                                                                               
-    mUI.progressBar->setRange(0, xccdf_policy_get_selected_rules_count(policy));                                                                                                              
+    mUI.progressBar->setRange(0, std::max(xccdf_policy_get_selected_rules_count(policy), 1));                                                                                                 
     mUI.progressBar->reset();                                                                                                                                                                 
     mUI.progressBar->setValue(0);
     mUI.progressBar->setEnabled(true);
@@ -1036,6 +1036,7 @@ void MainWindow::scanEnded(bool canceled)
     }
     else
     {
+        mUI.progressBar->setValue(mUI.progressBar->maximum());
         mUI.resultViewer->loadContent(mScanner);
         mUI.offlineRemediateButton->setEnabled(mScanner->getScannerMode() == SM_SCAN);
     }
```

**Comment by WesleyCeraso** [comment]
This issue was fixed by version 1.1 (3f5513ed)
